kratos/includes/properties.h

```diff
@@ -91,16 +86,33 @@ class Properties : public IndexedObject
 
     typedef std::unordered_map<std::size_t, TableType> TablesContainerType; // This is a provisional implmentation and should be changed to hash. Pooyan.
 
+    /// Properties container. A vector set of properties with their Id's as key.
+    typedef PointerVectorSet<Properties, IndexedObject> PropertiesContainerType;
+
+    /** Iterator over the properties. This iterator is an indirect
+    iterator over Properties::Pointer which turn back a reference to
+    properties by * operator and not a pointer for more convenient
+    usage. */
+    typedef typename PropertiesContainerType::iterator PropertiesIterator;
+
+    /** Const iterator over the properties. This iterator is an indirect
+    iterator over Properties::Pointer which turn back a reference to
+    properties by * operator and not a pointer for more convenient
+    usage. */
+    typedef typename PropertiesContainerType::const_iterator PropertiesConstantIterator;
 
     ///@}
     ///@name Life Cycle
     ///@{
 
     /// Default constructor.
-    Properties(IndexType NewId = 0) : BaseType(NewId), mData(), mTables() {}
+    Properties(IndexType NewId = 0) : BaseType(NewId), mData(), mTables(), mSubPropertiesList() {}
```

Comment:
mmm, these are also good candidates to be made explicit...

maybe as a part of another PR

Reply:
Preferably :)